src/main/java/internet/market/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package internet.market.service.impl;
+
+import internet.market.dao.ShoppingCartDao;
+import internet.market.lib.Inject;
+import internet.market.lib.Service;
+import internet.market.model.Product;
+import internet.market.model.ShoppingCart;
+import internet.market.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shoppingCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCartDao.getByUserId(shoppingCart.getUserId()).get().getProducts().add(product);
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        return shoppingCartDao.getByUserId(shoppingCart.getUserId())
+                .get().getProducts().remove(product);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCartDao.getByUserId(shoppingCart.getUserId()).get().getProducts().clear();
```

Comment:
You could probably have just 
`shoppingCartDao.getProducts().clear();`
here.

Reply:
it just for checking "does exsist this shoppingCart?" in case not method throws exception